### ISSUE TITLE
Screen showing state not saved if it is shown from another screen

### ISSUE THREAD

**Issue by ronchonbon** [renpy/renpy#5997]
In the following, if you click on button A to show screen A and then save and reload, screen A is hidden upon reload.

```
label start:
    show screen master_screen()

    while True:
        pause

    return

screen master_screen():
    textbutton "A" anchor (0.5, 0.5) pos (0.25, 0.5):
        action [
            Show("screen_a"),
            Hide("screen_b")]

    textbutton "B" anchor (0.5, 0.5) pos (0.75, 0.5):
        action [
            Hide("screen_a"),
            Show("screen_b")]

screen screen_a():
    text "SCREEN A SHOWN":
        align (0.5, 0.5)

screen screen_b():
    text "SCREEN B SHOWN":
        align (0.5, 0.5)
```
        
I can't tell if this is expected behavior or not - per the [save/load documentation](https://www.renpy.org/doc/html/save_load_rollback.html), screens being shown should be saved. I have tried putting renpy.checkpoint() and renpy.retain_after_load() in a couple places to no effect.

This is a minimal example of a more complicated setup where I have a master screen controlling access to and visibility of a bunch of other screens, with the master screen persistently shown during normal gameplay.

**Comment by renpytom** [comment]
Ren'Py is working correctly. 

When Ren'Py saves, it does so at the start of the current statement, to ensure that the game is in a consistent state. This means that if you show a screen, the fact of that screen being show isn't saved until you start the next statement. If you click through once, you'll find things save properly. 

It seems like renpy.retain_after_load would fix this, but that only deals with data, not what's showing on the screen. And that's a good thing, as it means that screens aren't shown more than once. 

Basically, think about when a script statement starts. Those are the only points at which Ren'Py can save.

**Comment by ronchonbon** [comment]
I see! I guess I'm working against the grain a bit - I'm working in a sandbox context where a player can potentially interact with a bunch of screens before getting to the next script statement, and it's not ideal for rollback and save/load to undo everything the player has done up until that point.

Based on your comment and some feedback on the Discord, this seems to achieve the desired effect even if it's a little ugly.

```
default dummy = 0

label start:
    show screen master_screen()

    # rest of game
    while True:
        pause

    return

label force_save_screen:
    $ dummy = 1

    return

screen master_screen():
    textbutton "A" anchor (0.5, 0.5) pos (0.25, 0.5):
        action [
            Hide("screen_b"),
            Show("screen_a"),
            Call("force_save_screen", from_current = True)]

    textbutton "B" anchor (0.5, 0.5) pos (0.75, 0.5):
        action [
            Hide("screen_a"),
            Show("screen_b"),
            Call("force_save_screen", from_current = True)]

screen screen_a():
    text "SCREEN A SHOWN":
        align (0.5, 0.5)

screen screen_b():
    text "SCREEN B SHOWN":
        align (0.5, 0.5)
```

At any rate, I think that resolves this!
